torch/_inductor/kernel/mm.py

```diff
@@ -107,7 +107,14 @@
             b = tl.load(B, mask=rk[:, None] < k, other=0.)
         if B_PROLOGUE_CAST_TYPE is not None:
             b = b.to(B_PROLOGUE_CAST_TYPE)
-        acc += tl.dot(a, b, allow_tf32=ALLOW_TF32)
+        input_precision = "tf32" if ALLOW_TF32 else "ieee"
+        if (
+            a.dtype.name == "float32" 
+            and b.dtype.name == "float32"
+            and {torch._inductor.config.triton.use_tf32x3}
```

Comment:
Inlining arbitrary config like this probably won't work? Seems templates are rendered from a limited env [here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/select_algorithm.py#L491-L495). Should we check the config outside the template and pass the extended argument(s) as per @Chillee's suggestion above?